docs/testing/e2e-plan.md

```diff
@@ -0,0 +1,48 @@
+# Curated Packages End-To-End Testing Plan
+
+This document summarizes the current state and future plans for end-to-end (E2E) testing in the EKS-A Curated Packages (CP) project.
```

Comment:
(Personal opinion) I prefer just "packages" instead of "CP".

Reply:
I think I'm going to try sticking with CP, because it's more explicit, and shorter. Package feels like such an overloaded term to begin with, and I don't think any of us will stop using it, but the explicitness of CP feels clearer for documentation.